yargs.js

```diff
@@ -912,6 +912,11 @@ function Yargs (processArgs, cwd, parentRequire) {
     enumerable: true
   })
 
+  self.then = function (onFulfilled, onRejected) {
+    return new Promise(resolve => resolve(self.argv))
```

Comment:
a couple suggestions to make this feature really rock:

1. I think we should run the `self.argv` in such a way that we catch the error (I would look at the approach that we use with `.parse()`) so that we can reject or resolve the promise depending on whether an error has occurred.
2. I think we should make it so (as with `.parse()`) we suppress output if an `onFulfilled` handler is provided.

having written these two points ... I definitely do think that I'd like to deprecate the `.parse(function())` behavior with this behavior -- I like this much better 👍